pkg/apis/core/validation/validation.go

```diff
@@ -128,7 +128,7 @@ func ValidateHasLabel(meta metav1.ObjectMeta, fldPath *field.Path, key, expected
 
 // ValidateAnnotations validates that a set of annotations are correctly defined.
 func ValidateAnnotations(annotations map[string]string, fldPath *field.Path) field.ErrorList {
-	return apimachineryvalidation.ValidateAnnotations(annotations, fldPath)
+	return apimachineryvalidation.ValidateAnnotations(annotations, fldPath).WithOrigin("format=k8s-key-label")
```

Comment:
Two things here:

1) "k8s-label-key"  and not  "k8s-key-label"

2) This needs to go INSIDE of `apimachineryvalidation.ValidateAnnotations()` -- that function validates more than one thing, and the way you have it now would attribute the "too long" error to label keys, which is not correct.

But I see you did add it inside the function, so we don't need an origin here at all - it will just corrupt the intention as expressed in the deeper function